src/supabase/sql/db_rls_settings.sql

```diff
@@ -1 +1,188 @@
--- TODO: RLSの設定を行う
\ No newline at end of file
+ALTER TABLE "User" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "UserFollower" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "UserLink" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "Recipe" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "RecipeImage" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "RecipeLink" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "Ingredient" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "Instruction" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "CartList" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "CartListItem" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "DraftRecipe" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "DraftRecipeImage" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "DraftRecipeLink" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "DraftIngredient" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "DraftInstruction" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "Favorite" ENABLE ROW LEVEL SECURITY;
+
+ALTER TABLE "Memo" ENABLE ROW LEVEL SECURITY;
+
+-- User
+
+CREATE POLICY "ユーザー情報取得" ON "public"."User" AS PERMISSIVE FOR
+SELECT
+    TO authenticated USING ( (auth.uid() = (user_id):: uuid))
+
+CREATE POLICY "ユーザー情報更新" ON "public"."User" AS PERMISSIVE FOR
+UPDATE
+    TO authenticated USING ( (auth.uid() = (user_id):: uuid))
+WITH
+    CHECK ( (auth.uid() = (user_id):: uuid))
+
+CREATE POLICY "ユーザー情報削除" ON "public"."User" AS PERMISSIVE FOR
+DELETE
+WITH
+    CHECK ( (auth.uid() = (user_id):: uuid))
+
+-- UserFollower
+
+CREATE POLICY "ユーザーのフォロー処理" ON "public"."UserFollower" AS PERMISSIVE FOR ALL TO authenticated USING (true)
+
+-- UserLink
+
+CREATE POLICY "ユーザーリンクの取得(全員が参照可能)" ON "public"."UserLink" AS PERMISSIVE FOR
+SELECT TO public USING (true)
+
+CREATE POLICY "ユーザーリンク作成" ON "public"."UserLink" AS PERMISSIVE FOR
+INSERT TO authenticated
+WITH
+    CHECK ( (auth.uid() = (user_id):: uuid))
+
+CREATE POLICY "ユーザーリンク更新" ON "public"."UserLink" AS PERMISSIVE FOR
+UPDATE
+    TO authenticated USING ( (auth.uid() = (user_id):: uuid))
+WITH
+    CHECK ( (auth.uid() = (user_id):: uuid))
+
+-- Recipe
+
+CREATE POLICY "レシピ情報取得(公開・非公開の設定はコード" ON "public"."Recipe" AS PERMISSIVE FOR
+SELECT TO public USING (true)
+
+CREATE POLICY "レシピ情報作成" ON "public"."Recipe" AS PERMISSIVE FOR
+INSERT
+WITH
+    CHECK ( (auth.uid() = (user_id):: uuid))
+
+CREATE POLICY "レシピ情報更新" ON "public"."Recipe" AS PERMISSIVE FOR
+UPDATE
+    TO authenticated USING ( (auth.uid() = (user_id):: uuid))
+
+CREATE POLICY "レシピ情報削除" ON "public"."Recipe" AS PERMISSIVE FOR
+DELETE
+    TO authenticated USING ( (auth.uid() = (user_id):: uuid))
```

Comment:
シェフのレシピは管理者が作成するので管理者権限を持つユーザーも削除できるように以下のように改良する必要があると思います
作成や更新も同様ですかね

```suggestion
CREATE POLICY "レシピ情報削除" ON "public"."Recipe" AS PERMISSIVE FOR
DELETE
USING (user.role = 'ADMIN' OR auth.uid() = user_id)
```